contracts/cw3-flex-multisig/src/contract.rs

```diff
@@ -1420,7 +1426,7 @@ mod tests {
                 &[],
             )
             .unwrap_err();
-        assert_eq!(ContractError::Unauthorized {}.to_string(), err);
+        assert_eq!(ContractError::Unauthorized {}, err.downcast().unwrap());
```

Comment:
Downcast is basically generic with signature `fn downcast<T: Any>(self) -> T;`, and you can call it just like `err.downcast::<Error>()`. Here it is just type elision - `assert_eq` internally uses `PartiallEq`, so type elision checks all `PartiallEq` implementations of `ContractError`. It turns out that there is only one such implementation, so right side have to be of proper type (here `ContractError`).

Note - because of how it worsk, the downcasted error have to be on the righ side of `==` - because left-side `PartialEq` implementation is used. If error is on left side, type elision need to solve query "Find any type which have implementation of `PartialEq<ContractError>`" which is semantically impossible, as anyone may add such implementation.

Reply:
Ah, I see it does runtime checks...

I tried:

```rust
        let unwrapped = err.downcast::<&str>().unwrap();
        assert_eq!("unauthorized", unwrapped);
```

and got

```
thread 'contract::tests::execute_group_changes_from_proposal' panicked at 'called `Result::unwrap()` on an `Err` value: Unauthorized', contracts/cw3-flex-multisig/src/contract.rs:1429:48
```

Anyway, makes better sense to me how this works in practice. Code is fine, just my learning the new lib.